### PR TITLE
Reporting the number of tests in GenotypePhenotypeAnalysisResult

### LINKED ISSUE #98: Report the number of performed tests in `GenotypePhenotypeAnalysisResult`
When doing multiple tests, knowing how many tests were preformed is a useful information to have.

We should provide the users with a simple way to get the number of performed tests.

This can be implemented as a new `int` property on `GenotypePhenotypeAnalysisResult`.

### PR DESCRIPTION
**Pull request by lnrekerle**
This will close #98 

Added a property to GenotypePhenotypeAnalysisResult called total_tests and it reports the total number of phenotype tests that were run for that analysis.

### CODE REVIEW

**Review by pnrobinson — DISMISSED** [review]

**Comment by pnrobinson** [comment]
@lnrekerle please clean up the CI: AttributeError: 'GpCohortAnalysis' object has no attribute 'compare_by_recessive_variant_effect'

File: src/genophenocorr/analysis/_gp_impl.py [hunk 1]
```diff
@@ -47,42 +47,6 @@ def __init__(
...
-    def compare_by_recessive_protein_feature(self, feature: str, tx_id: str) -> GenotypePhenotypeAnalysisResult:
-        predicate = self._protein_predicates.protein_feature(feature_id=feature, tx_id=tx_id)
-        recessive_predicate = wrap_as_recessive_predicate(predicate)
-        return self._apply_poly_predicate_on_hpo_terms(recessive_predicate)
-    
-    # TODO: remove
-    def compare_by_recessive_protein_region(self, region:Region, tx_id:str) -> GenotypePhenotypeAnalysisResult:
-        predicate = VariantPredicates.region(region=region, tx_id=tx_id)
-        recessive_predicate = wrap_as_recessive_predicate(predicate)
-        return self._apply_poly_predicate_on_hpo_terms(recessive_predicate)
```
Comment (ielis) [inline src/genophenocorr/analysis/_gp_impl.py:50]: I concur on the need of removing this at some point soon, but it isn't 100% necessary to do it in this PR, unless you take the time to remove the usages across the entire code base (+notebooks).

Pls either revert the methods or remove all usages.

**Review by ielis — CHANGES_REQUESTED** [review]
Looks good except for the one issue below (also pointed out by Peter).

**Review by ielis — APPROVED** [review]
Looks great, thank you!

**Pull request merged**